src/item_group.cpp

```diff
@@ -191,7 +191,12 @@ void Item_modifier::modify(item &new_item) const
             // spawn a "water (0)" item.
             new_item.charges = std::max( 1l, ch );
         } else if(t != NULL) {
-            new_item.charges = std::min(ch, t->max_charges);
+            const auto a = new_item.ammo_type();
```

Comment:
you can probably just repeat the call to `ammo_type()` without defining a scoped variable. any sane compiler will generate identical code and there isn't a more descriptive variable name that you could use here